day3/recipe/RecipesModel.h

```diff
@@ -0,0 +1,59 @@
+#ifndef RECIPESMODEL_H
+#define RECIPESMODEL_H
+
+#include <QtCore>
+#include <QtGui>
+#include <memory>
+
+
+class RecipesModel : public QAbstractListModel
+{
+    Q_OBJECT
+public:
+    enum DifficultyLevel {
+        LowDifficulty,
+        MidDifficulty,
+        HighDifficulty,
+        DifficultyCount,
+    };
+    Q_ENUM(DifficultyLevel)
+
+    static const QString difficultyStrings[];
+
+    enum RoleNames {
+        DescriptionRole = Qt::UserRole,
+        ImageSourceRole = Qt::UserRole + 1,
+        DifficultyRole = Qt::UserRole + 2,
+        PreparationTimeRole = Qt::UserRole + 3,
+        DifficultyStringRole = Qt::UserRole + 4,
+        PreferredRole = Qt::UserRole + 5
+    };
+
+    struct RecipeData {
+        RecipeData(QString descr, QString  imageSo, DifficultyLevel difficul, int prepTime, bool pref): description(descr), imageSource(imageSo), difficulty(difficul)
+            , preparationTime(prepTime), preferred(pref){}
+
+        QString  description;
+        QString  imageSource;
+        DifficultyLevel difficulty;
+        int preparationTime;
+        bool preferred;
+    };
+
+    explicit RecipesModel(QObject *parent = 0);
+    ~RecipesModel();
```

Comment:
Non ti far fregare dell'IDE che crea il distruttore vuoto! :smile: 
Puoi cancellare l'implementazione nel `.cpp` se usi `= default`
```suggestion
    ~RecipesModel() = default;
```